components/gitpod-db/src/typeorm/migration/1715064681193-DeleteDanglingProjectEnvVars.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class DeleteDanglingProjectEnvVars1715064681193 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(
+            "DELETE FROM d_b_project_env_var WHERE projectId IN (SELECT id FROM d_b_project WHERE markedDeleted=true)",
```

Comment:
Curious: Did you check the complexity/runtime of this query?

Reply:
I executed it with a SELECT against prod DB and it returns immediately